### ISSUE TITLE
Exit code detection failing on iOS/tvOS 15+

### ISSUE THREAD

**Issue by premun** [dotnet/xharness#819]
With the latest move of `dotnet/runtime` onto OSX 12.00, `apple run` and `apple just-run` commands started failing in 100% of cases:

![image](https://user-images.githubusercontent.com/7013027/153199265-413daa78-e834-4b01-85c8-51b205c654a9.png)

```sql
Metrics
| where EventType == "_MobileDeviceOperation" and MetricName == "ExitCode"
| join kind=inner Jobs on JobId
| where Finished > now()-1d
| extend Dimensions = parse_json(Dimensions)
| project 
    Finished,
    JobName,
    QueueName,
    WorkItemFriendlyName,
    Platform = tostring(Dimensions.platform),
    Command = tostring(Dimensions.command),
    ExitCode = MetricValue,
    Target = tostring(Dimensions.target),
    IsDevice = tobool(Dimensions.isDevice)
| where Platform == "apple"
| summarize SuccessRate=countif(ExitCode < 6)*100/count(), Count=count() by Platform, Command, QueueName
| where SuccessRate != 100
| order by Platform, SuccessRate, Command
```

**Comment by premun** [comment]
Some reading for later
- https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man2/kqueue.2.html
- https://stackoverflow.com/a/24690619/6289970
- https://developer.apple.com/library/archive/technotes/tn2050/_index.html

**Comment by premun** [comment]
Seems like the issue is Xcode 13.2 rather than OSX 12.00.

Also I found out, I might be able to use `kqueue` and `kevent` to get notified about a process' exit where the exit code should be in the data of the event:
https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man2/kevent.2.html

```
EVFILT_PROC  Takes the process ID to monitor as the	identifier and
			 the events to watch for in fflags, and	returns	when
			 the process performs one or more of the requested
			 events.  If a process can normally see	another
			 process, it can attach	an event to it.	 The events to
			 monitor are:

			 NOTE_EXIT	   The process has exited. The exit status will be stored in data.
```

**Comment by premun** [comment]
Oh noez, the sentence _"The exit status will be stored in data."_ actually only appears in FreeBSD man pages and not OSX man pages 😢

**Comment by premun** [comment]
I was able to get more logs via:
```
xcrun simctl spawn booted log stream --level=Info | grep [APP NAME]
```

The exit code is there, so is the stdout. So we should also be able to fix #694

**Comment by premun** [comment]
Solution for Simulators is here: https://github.com/premun/xharness/tree/prvysoky/simulator-exit-code

MacCatalyst is left
